documentation/proposals/Proposal - Axis Input Devices.md

```diff
@@ -0,0 +1,624 @@
+    Point = 1 << 8 | Analog | HasRawValue,
+
+    // ---- Physical orientation ----
+    Orientation = 1 << 2 | Analog,
+    Rotation = 1 << 3 | Orientation,
+
+    // this is specified because single axes of euler angle components can be useful by themselves,
+    // whereas anything shorter than 3 or 4 axes is not for quaternion rotation components
+    /// <summary>
+    /// Indicates that this axis is a component of a euler angle (do we want this to be in degrees or radians?)
+    /// If this flag is set, the axis is expected to be normalized between 0-360 degrees, with wrapping applied
+    /// It is required you do NOT set <see cref="AxisDescription.RawValueBounds"/>, as it will be assumed to be (0, 360/2pi).
+    /// </summary>
+    EulerAngleComponent = 1 << 4 | Rotation | HasRawValue,
+
+    /// <summary>
+    /// Acceleration - in meters per second <br/>
+    /// If this is set, it is expected that the axis is NOT provided any raw value constraints,
+    /// as it is not expected to be normalized
+    /// </summary>
+    Acceleration = 1 << 6 | Orientation | RawValueOnly,
+
+    /// <summary>
+    /// Position - raw values are in meters if world-space, in pixels if screen-space, etc <br/>
+    /// Normalized values are required if the bounds are known via the device's API.
+    /// If they are unknown, this must be paired with <see cref="RawValueOnly"/>
+    /// </summary>
+    Position = 1 << 7 | Orientation | Point,
+    // -----------------------------
+
+    /// <summary>
+    /// Indicates that the raw value provided for this axis is not the same as the normalized value. <br/>
+    /// If this flag is signaled on an axis, it is highly recommended that the bounds are provided as well, even if they are infinite
+    /// This can be useful for things like euler angle components, where single axes 0-360 degrees (or 0-2pi radians? do we want to standardize which?) can be useful by themselves
+    /// </summary>
+    HasRawValue = 1 << 24 | Analog,
+
+    /// <summary>
+    /// Indicates that there is no one-size-fits-all way to normalize this axis, and must be handled in a case-by-case basis <br/>
+    /// Examples of this would be: <br/>
+    /// - Accelerometers, as the relevant magnitude differs depending on the application <br/>
+    /// - VR controller position without a bounding area <br/>
+    /// - Quaternion rotation values, as these can be expected to be negative <br/>
+    /// <br/>
+    /// If this is set, it is expected that the axis is NOT provided any raw value constraints
+    /// </summary>
+    RawValueOnly = 1 << 25 | HasRawValue,
+
+    /// <summary>
+    /// Things like a mouse movement, trackpad, etc
+    /// </summary>
+    Delta = 1 << 26 | HasRawValue,
+
+    /// <summary>
+    /// Indicates that there is no one-size-fits-all way to normalize this axis, and must be handled in a case-by-case basis <br/>
+    /// Examples of this would be: <br/>
+    /// - Mouse position <br/>

```

Comment:
We probably need to look more into how we map an axis device into a pointer device for instance, I'd imagine a lot of questions I may have will naturally occur in trying to define this. (trying to think of the pointer targets and the different possible bounds)

Reply:
I think all you need to worry about is the fact that this API will provide you with a normalized 1d, 2d, 3d position value from the input device, as well as any relevant rotation information. projecting that into any different bounds should be fairly straightforward from there as a mapping function if a user wants to hone in on a subset of a device surface, no?

Reply:
It’s more the transitioning between bounded vs unbounded for which we have the IPointerTargets. Basically this API is just observation of current state only (which is all you need most of the time) and there’s no way to set anything for a device.